src/Secure/OpenIdConnectAuthentication.php

```diff
@@ -59,13 +62,20 @@ protected function getCluster(): ?string
 
     protected function getSoapHeaders()
```

Comment:
Perhaps a `OpenIdConnectAuthentication::setOffice` method could be added, such that you can always set the Office afterwards. This method would then have to set `AuthenticatedConnection::$authenticatedClients` to an empty array, to make sure new clients (with the `CompanyCode` header) will be created afterwards.